stetho-sample/src/debug/java/com/facebook/stetho/sample/CalendarProviderDatabasePeer.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.sample;
+
+import android.annotation.TargetApi;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteException;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.CalendarContract;
+import com.facebook.stetho.inspector.jsonrpc.JsonRpcPeer;
+import com.facebook.stetho.inspector.protocol.module.Database;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is example of extending database domain by adding calendar provider
+ */
+public class CalendarProviderDatabasePeer extends Database.DatabasePeer {
+
+  private final static String sDatabaseName = "provider-calendars";
+
+  public CalendarProviderDatabasePeer(Context context) {
+    super(context);
+  }
+
+  @Override
+  protected void onRegistered(JsonRpcPeer peer) {
+    Database.DatabaseObject databaseParams = new Database.DatabaseObject();
+    databaseParams.id = sDatabaseName;
+    databaseParams.name = sDatabaseName;
+    databaseParams.domain = mContext.getPackageName();
+    databaseParams.version = "N/A";
+    Database.AddDatabaseEvent eventParams = new Database.AddDatabaseEvent();
+    eventParams.database = databaseParams;
+    peer.invokeMethod("Database.addDatabase", eventParams, null /* callback */);
+  }
+
+  @Override
+  protected void onUnregistered(JsonRpcPeer peer) {
+
+  }
+
+  @Override
+  public boolean contains(String databaseId) {
+    return sDatabaseName.equals(databaseId);
+  }
+
+  @Override
+  public List<String> getDatabaseTableNames(String databaseId) {
+    List<String> tableNames = new ArrayList<String>();
+    tableNames.add("calendars");
+    tableNames.add("events");
+    return tableNames;
+  }
+
+  @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+  @Override
+  public <T> T executeSQL(String databaseName, String query, Database.DatabasePeer.ExecuteResultHandler<T> handler) throws SQLiteException {
+
+    // Resolve query and make the right Cursor. Some better query parser should exist. But for now it's fine.
+    Uri uri = null;
+    String[] projection = null;
+    if (query.contains("calendars")) {
```

Comment:
Would be really great if we provided a helper that lets you declarative the schema and then just let a generic driver operate on that schema.

For instance, something like:
```java
new ContentProviderSchema.Builder(...)
   .table(new Table.Builder()
       .uri(...)
       .projection(...) // optional
       .build())
   .build();
```

Then you could have an abstract `ContentProviderDatabaseDriver` that takes in the schema.